File: oldexams/bakery.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;
int minValues[100001][3] = {0};

bool binarySearch(int *array,int key,int first,int last){
    if(first > last)
        return false;
    int middle = (first+last)/2;
    if(array[middle] < key)
        return binarySearch(array,key,middle+1,last);
    if(array[middle] > key)
        return binarySearch(array,key,first,middle-1);
    else
        return true;
}

int main()
{
    int n,m,k;
    cin >> n >> m >> k;
    for (int i = 0; i < m; i++){
        int x,y,z;
        cin >> x >> y >> z;

        minValues[i][0] = x;
        minValues[i][1] = y;
        minValues[i][2] = z;
    }
    int *storages = new int[k];
    for(int i = 0; i < k; i++)
        cin >> storages[i];
    if(k == 0){
            cout << -1;
            return 0;
    }
    int minimum = -1;
    sort(storages,storages+k);
    for(int i = 0; i < m; i++){
        int temp;
        if((binarySearch(storages,minValues[i][0],0,k) && !(binarySearch(storages,minValues[i][1],0,k)))||(!(binarySearch(storages,minValues[i][0],0,k)) && (binarySearch(storages,minValues[i][1],0,k)))){
            temp = minValues[i][2];
            if(minimum > temp || minimum == -1)
                minimum = temp;
        }
    }
    cout << minimum;
    return 0;

}

File: oldexams/Bakery.py
def binarySearch(alist,item):
    first = 0
    last = len(alist)- 1
    found = False
    while first <= last and not found:
        middle = (first+last)/2
        if alist[middle] == item:
            fount = True
        else:
            if item < alist[middle]:
                last = middle-1
            else:
                first = middle+1
    return found

n,m,k = map(int,raw_input().strip().split())
arr = [[-1 for a in range(n+1)] for a in range(n+1)]
harita = []
for a in range(m):
    x,y,z = map(int,raw_input().strip().split())
    if arr[x][y] != -1 and arr[x][y] > z:
        arr[x][y] = z
    harita.append([x,y,z])
if k:
    arr2 = map(int,raw_input().strip().split())
    arr2.sort()
    temp = sorted(arr,key= lambda second:arr[2])
    minimum= -1
    for a in range(m):
        if ((binarySearch(arr2,harita[a][0])) and not(binarySearch(arr2,harita[a][1]))) or (not(binarySearch(arr2,harita[a][0])) and (binarySearch(arr2,harita[a][1]))):
            if minimum == -1 or minimum > harita[2]:
                minimum = harita[2]
    print minimum
if not k:
    print -1
